common-content/en/module/tools/comparing-javascript-and-python/index.md

```diff
@@ -0,0 +1,68 @@
++++
+title = "Comparing JavaScript and Python"
+headless = true
+time = 40
+facilitation = false
+emoji= "📖"
+objectives = [
+    "Identify and explain equivalences between JavaScript and Python",
+    "Identify and explain differences between JavaScript and Python",
+    "Distinguish between essential and accidental complexity"
+]
++++
+
+JavaScript and Python are quite similar languages in a lot of ways.
+
+Most of the differences between them are quite cosmetic. e.g.
+* Some functions and operators have different names. But often there are functions/operators which do exactly the same thing.
+* JavaScript uses `{}` around blocks of code and we optionally _choose_ to indent code, whereas Python uses `:` and _required_ indents.
+* In JavaScript we choose to name variables in `camelCase`, whereas in Python we choose to name variables in `snake_case` (but in both langues we _could_ do either).
+
+Let's take our "count containing words" JavaScript code from last week, and think about what it would look like in Python.
+
+```js
+import { program } from "commander";
+import { promises as fs } from "node:fs";
+import process from "node:process";
+
+program
+    .name("count-containing-words")
+    .description("Counts words in a file that contain a particular character")
+    .option("-c, --char <char>", "The character to search for", "-");
+
+program.parse();
+
+const argv = program.args;
+if (argv.length != 1) {
+    console.error(`Expected exactly 1 argument (a path) to be passed but got ${argv.length}.`);
+    process.exit(1);
+}
+const path = argv[0];
+const char = program.opts().char;
+
+const content = await fs.readFile(path, "utf-8");
+const wordsContainingChar = content.split(" ").filter((word) => word.indexOf(char) > -1).length;
+console.log(wordsContainingChar);
+```
+
+Let's think about what we're doing in this code. We're:
+* Parsing command line flags - writing down what flags we expect to be passed, and reading values for them based on the actual command line.
```

Comment:
Make this an activity and put the answer in a tab. Get them to write it out first themselves